h2o-core/src/main/java/water/util/LogArchiveWriter.java

```diff
@@ -0,0 +1,104 @@
+package water.util;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Date;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Abstract layer over different types of log containers (zip, txt,...)
+ * @param <OS>
```

Comment:
NIT: doc for `<OS>` could be a but more explanatory (or removed)

Reply:
yeah, will fix - this is ugly anyway